### PR TITLE
Added Ubuntu 18.04 to CI and refactored Jenkinsfile.

### PR DESCRIPTION
**Pull request by johnkord**

### CODE REVIEW

**Comment by johnkord** [comment]
This adds two clang-7 tests for Ubuntu 18.04 into our CI. We should eventually support more configurations!

It also refactors and renames a few things in the Jenkinsfile to use slightly different language.

**Comment by johnkord** [comment]
bors try

**Comment by johnkord** [comment]
bors try

File: .jenkins/Jenkinsfile [hunk 1]
```diff
@@ -159,19 +159,21 @@ def windowsReleaseCrossPlatform() {
...
-        "Coffeelake clang-7 SGX1-FLC Debug" : { coffeelakeTest('clang-7', 'SGX1FLC', 'Debug') },
-        "Coffeelake clang-7 SGX1-FLC Release" : { coffeelakeTest('clang-7', 'SGX1FLC','Release') },
-        "Coffeelake clang-7 SGX1-FLC RelWithDebInfo" : { coffeelakeTest('clang-7', 'SGX1FLC', 'RelWithDebinfo') },
-        "Coffeelake gcc SGX1-FLC Debug" : { coffeelakeTest('gcc', 'SGX1FLC', 'Debug') },
-        "Coffeelake gcc SGX1-FLC Release" : { coffeelakeTest('gcc', 'SGX1FLC', 'Release') },
-        "Coffeelake gcc SGX1-FLC RelWithDebInfo" : { coffeelakeTest('gcc', 'SGX1FLC', 'RelWithDebInfo') },
-        "Windows Debug Cross-platform" : { windowsDebugCrossPlatform() },
-        "Windows Release Cross-platform" : { windowsReleaseCrossPlatform() },
-        "Non-Simulation Container SGX1-FLC RelWithDebInfo" : { nonSimulationTest() }
+parallel "Check Pre-Commit Requirements" :      { checkPreCommitRequirements() },
```
Comment (andschwa) [inline .jenkins/Jenkinsfile]: What a nice matrix!

File: .jenkins/Jenkinsfile [hunk 2]
```diff
@@ -159,19 +159,21 @@ def windowsReleaseCrossPlatform() {
...
+        "Sim clang-7 SGX1-FLC RelWithDebInfo" : { simulationTest('clang-7', 'SGX1FLC', 'RelWithDebInfo')},
+        "ACC1604 clang-7 Debug" :               { ACCTest('ACC-1604', 'clang-7', 'SGX1FLC', 'Debug') },
+        "ACC1604 clang-7 Release" :             { ACCTest('ACC-1604', 'clang-7', 'SGX1FLC','Release') },
+        "ACC1604 clang-7 RelWithDebInfo" :      { ACCTest('ACC-1604', 'clang-7', 'SGX1FLC', 'RelWithDebinfo') },
+        "ACC1604 gcc Debug" :                   { ACCTest('ACC-1604', 'gcc', 'SGX1FLC', 'Debug') },
+        "ACC1604 gcc Release" :                 { ACCTest('ACC-1604', 'gcc', 'SGX1FLC', 'Release') },
+        "ACC1604 gcc RelWithDebInfo" :          { ACCTest('ACC-1604', 'gcc', 'SGX1FLC', 'RelWithDebInfo') },
+        "ACC1804 clang-7 Debug" :               { ACCTest('ACC-1804', 'clang-7', 'SGX1FLC', 'Debug') },
+        "ACC1804 clang-7 RelWithDebInfo" :      { ACCTest('ACC-1804', 'clang-7', 'SGX1FLC', 'RelWithDebinfo') },
+        "Win2016 Debug Cross-platform" :        { win2016DebugCrossPlatform() },
```
Comment (soccerGB) [inline .jenkins/Jenkinsfile]: Is it on purpose to have only two flavors for 1804?

Comment (andschwa) [reply .jenkins/Jenkinsfile]: > This adds two clang-7 tests for Ubuntu 18.04 into our CI. We should eventually support more configurations!

I think so!

**Comment by johnkord** [comment]
Partially, yes. One reason is that I didn't have time to set up another ACC system to support quick enough overall job builds and I don't want this one system to be a bottleneck. So thats a good opportunity for Cloudbase to test out their cross platform ansible setup playbooks for CI when they create another 1804 system.

Another reason is that I'd like to enable some kind of builds on 1804 in CI asap so we don't break it

**Pull request merged**